deepchem/models/torch_models/gcn.py

```diff
@@ -0,0 +1,307 @@
+        except:
+            raise ImportError('This class requires dgl.')
+        try:
+            import dgllife
+        except:
+            raise ImportError('This class requires dgllife.')
+
+        if mode not in ['classification', 'regression']:
+            raise ValueError("mode must be either 'classification' or 'regression'")
+
+        super(GCN, self).__init__()
+
+        self.n_tasks = n_tasks
+        self.mode = mode
+        self.n_classes = n_classes
+        self.nfeat_name = nfeat_name
+        if mode == 'classification':
+            out_size = n_tasks * n_classes
+        else:
+            out_size = n_tasks
+
+        from dgllife.model import GCNPredictor as DGLGCNPredictor
+
+        if activation is not None:
+            activation = [activation] * num_gnn_layers
+
+        self.model = DGLGCNPredictor(in_feats=in_node_dim,
+                                     hidden_feats=[hidden_node_dim] * num_gnn_layers,
+                                     activation=activation,
+                                     residual=[residual] * num_gnn_layers,
+                                     batchnorm=[batchnorm] * num_gnn_layers,
+                                     dropout=[dropout] * num_gnn_layers,
+                                     n_tasks=out_size,
+                                     predictor_hidden_feats=predictor_hidden_feats,
+                                     predictor_dropout=predictor_dropout)
+
+    def forward(self, g):
+        """Predict graph labels
+
+        Parameters
+        ----------
+        g: DGLGraph
+            A DGLGraph for a batch of graphs. It stores the node features in
+            ``dgl_graph.ndata[self.nfeat_name]``.
+
+        Returns
+        -------
+        torch.Tensor
+            The model output.
+
+            * When self.mode = 'regression',
+              its shape will be ``(dgl_graph.batch_size, self.n_tasks)``.
+            * When self.mode = 'classification', the output consists of probabilities
+              for classes. Its shape will be
+              ``(dgl_graph.batch_size, self.n_tasks, self.n_classes)`` if self.n_tasks > 1;
+              its shape will be ``(dgl_graph.batch_size, self.n_classes)`` if self.n_tasks is 1.
+        torch.Tensor, optional
+            This is only returned when self.mode = 'classification', the output consists of the
+            logits for classes before softmax.
+        """
+        node_feats = g.ndata.pop(self.nfeat_name)
+        out = self.model(g, node_feats)
+
+        if self.mode == 'classification':
+            if self.n_tasks == 1:
+                logits = out.view(-1, self.n_classes)
+                softmax_dim = 1
+            else:
+                logits = out.view(-1, self.n_tasks, self.n_classes)
+                softmax_dim = 2
+            proba = F.softmax(logits, dim=softmax_dim)
+            return proba, logits
+        else:
+            return out
+
+class GCNModel(TorchModel):
```

Comment:
These new classes are great! Could you add them to the docs as well in `docs/model.rst` so they get rendered on readthedocs

Reply:
Done.